### PR TITLE
Add Puppet 6 support

### PR DESCRIPTION
**Pull request by seanmil**
## Overview

This pull request introduces Puppet 6 support for computed catalogs and fixes issue #186 .

This change is based on the information contained in that issue and contains the conditional handling to make this change non-disruptive with existing Octocatalog-diff uses on older versions. It also contains updates to the testing suite to include tests covering the changes as well as updates to the CI pipeline to introduce Puppet 6 into the testing matrix.

## Checklist

- [x] Make sure that all of the tests pass, and fix any that don't. Just run `rake` in your checkout directory, or review the CI job triggered whenever you push to a pull request.
- [x] Make sure that there is 100% [test coverage](https://github.com/github/octocatalog-diff/blob/master/doc/dev/coverage.md) by running `rake coverage:spec` or ignoring untestable sections of code with `# :nocov` comments. If you need help getting to 100% coverage please ask; however, don't just submit code with no tests.
- [x] If you have added a new command line option, we would greatly appreciate a corresponding [integration test](https://github.com/github/octocatalog-diff/blob/master/doc/dev/integration-tests.md) that exercises it from start to finish. This is optional but recommended.
- [x] If you have added any new gem dependencies, make sure those gems are licensed under the MIT or Apache 2.0 license. We cannot add any dependencies on gems licensed under GPL.
- [x] If you have added any new gem dependencies, make sure you've checked in a copy of the `.gem` file into the [vendor/cache](https://github.com/github/octocatalog-diff/tree/master/vendor/cache) directory.

### CODE REVIEW

**Comment by KlavsKlavsen** [comment]
Hoping this gets merged.. Fantastic work @seanmil - I was really hoping to see this tool work for recent puppet installations again :)

File: lib/octocatalog-diff/catalog-util/command.rb [hunk 1]
```diff
@@ -95,9 +107,16 @@ def setup
           --no-daemonize
           --no-ca
```
Comment (ashish1099) [inline lib/octocatalog-diff/catalog-util/command.rb:108]: Check on puppet 6.19 and `--no-ca` option is removed now

Comment (seanmil) [reply lib/octocatalog-diff/catalog-util/command.rb]: Good catch. I've added a conditional to only include the `--no-ca` option on Puppet versions < 6.0. Thanks!

**Comment by seanmil** [comment]
@ashish1099 I re-ran the CI job. It passed, but I'm not sure why the results above still show "Waiting for status to be reported". Please see https://github.com/seanmil/octocatalog-diff/actions/runs/415637932 for the results. If we could get this merged, that would be great. Thank you.

**Comment by ashish1099** [comment]
@seanmil hmm, I don't think I have any rights on this repo to merge this MR :)

**Comment by seanmil** [comment]
> @seanmil hmm, I don't think I have any rights on this repo to merge this MR :)

Oh, shucks, I thought you were with GitHub. Perhaps that was just my hope that someone there was paying attention to this project.

**Comment by ashish1099** [comment]
> > @seanmil hmm, I don't think I have any rights on this repo to merge this MR :)
> 
> Oh, shucks, I thought you were with GitHub. Perhaps that was just my hope that someone there was paying attention to this project.

But thanks for the PR, I have been using this branch for my testing setup 👍

**Review by ahayworth — APPROVED** [review]
🎉 🎉 🎉 🎉 🎉 
Amazing. I'm so happy to see this PR, and I'm so happy that it looks like it wasn't too much work (after Puppet added back the catalog compilation bits, of course).
🎉 🎉 🎉 🎉 🎉

**Comment by ashish1099** [comment]
next step would be a release of tag ? so I can just download the required tag.

currently I have to mess around in my docker image to get things running, but no rush 👍

**Comment by ahayworth** [comment]
Yes - I'm planning on doing a release sometime this week (or maybe as a Christmas present, who knows 🎄 🎅) with this and a few other PRs rolled into it. 😄

**Pull request merged**